### PR TITLE
Fix for OpenGL version selection and reporting on Windows and X

### PR DESCRIPTION
**Pull request by EdgarReynaldo**
I expanded my patches to include fixes for X.

### CODE REVIEW

**Comment by dos1** [comment]
I've just sent a similar patch for SDL backend, but as far as I can see, it's not handled by any other platform anyway...

File: src/win/wgl_disp.c [hunk 1]
```diff
@@ -1004,6 +1002,13 @@ static bool create_display_internals(ALLEGRO_DISPLAY_WGL *wgl_disp)
       disp->extra_settings.settings[ALLEGRO_COMPATIBLE_DISPLAY] = 0;
    }
 
+   /* Fill in the display settings for opengl major and minor versions...*/
+   int* s = disp->extra_settings.settings;
```
[the remark this answers is not in the document]
Comment (EdgarReynaldo) [reply src/win/wgl_disp.c]: Fixed to match xdisplay.c

File: src/win/wgl_disp.c [hunk 2]
```diff
@@ -959,10 +959,8 @@ static bool create_display_internals(ALLEGRO_DISPLAY_WGL *wgl_disp)
       return false;
    }
 
-   major = _al_get_suggested_display_option(disp,
-      ALLEGRO_OPENGL_MAJOR_VERSION, 0);
-   minor = _al_get_suggested_display_option(disp,
-      ALLEGRO_OPENGL_MINOR_VERSION, 0);
+   major = al_get_new_display_option(ALLEGRO_OPENGL_MAJOR_VERSION , 0);
```
[the remark this answers is not in the document]
Comment (EdgarReynaldo) [reply src/win/wgl_disp.c]: Spaces removed.

File: src/opengl/extensions.c [hunk 3]
```diff
@@ -798,7 +799,7 @@ void _al_ogl_manage_extensions(ALLEGRO_DISPLAY *gl_disp)
    /* Need that symbol already so can't wait until it is assigned later. */
    glGetStringi = ext_api->GetStringi;
 
-   if (_al_ogl_version_3_only(gl_disp->flags)) {
+   if (gl_disp->ogl_extras->ogl_info.version >= _ALLEGRO_OPENGL_VERSION_3_0) {
```
[the remark this answers is not in the document]
Comment (EdgarReynaldo) [reply src/opengl/extensions.c]: I changed that too. We shouldn't rely on the driver flags.

**Comment by SiegeLord** [comment]
That looks good, thanks!

**Pull request merged**